controllers/useraccount/useraccount_controller.go

```diff
@@ -167,8 +167,8 @@ func (r *Reconciler) ensureUserAndIdentity(logger logr.Logger, userAcc *toolchai
 	var createdOrUpdated bool
 	var user *userv1.User
 	var err error
-	// for AppStudio use-case - create User & Identity resources if user is not using appstudio tier
-	if userAcc.Spec.NSTemplateSet == nil || userAcc.Spec.NSTemplateSet.TierName != "appstudio" {
+	// for non-AppStudio use-case - create User & Identity resources if user is not using appstudio tier
+	if userAcc.Spec.NSTemplateSet == nil || (userAcc.Labels != nil && userAcc.Labels[toolchainv1alpha1.TierLabelKey] != "appstudio") {
```

Comment:
you don't need the first evaluation `userAcc.Spec.NSTemplateSet == nil` any more there, otherwise, when you do the migration it would provision `User` & `Identity` even for appstudio users (the `NSTemplateSet` will be nill for all UserAccounts). 
In addition, the default behavior should be the creation of the resources (`User` & `Identity`), not deleting them - the resources should not be present only and only when the tier label is `appstudio`, so when either the labels are nil or the label is not present, then it should follow the default procedure.
```suggestion
	if userAcc.Labels == nil || userAcc.Labels[toolchainv1alpha1.TierLabelKey] != "appstudio" {
```